### ISSUE TITLE
MD051 - files in repo/gist

### ISSUE THREAD

**Issue by anonymous1184** [DavidAnson/markdownlint#940]
The rule triggers files in the same repo/gist.

Example `README.md`:

![image](https://github.com/DavidAnson/markdownlint/assets/53758552/b7d20f8c-3f7d-4cfb-b957-95568449dc4a)

```md
[Self][1]

[1]: #file-readme-md
```

**Comment by DavidAnson** [comment]
That is correct, none of the automatically generated fragments in that file match "file-readme-md".

The rule documentation explains more: https://github.com/DavidAnson/markdownlint/blob/main/doc/md051.md

If you change the top level heading to "File README MD", the warning goes away because that creates a matching fragment: https://dlaa.me/markdownlint/#%25m%23%20File%20README%20MD%0A%0A%5BSelf%5D%5B1%5D%0A%0A%5B1%5D%3A%20%23file-readme-md%0A

**Comment by anonymous1184** [comment]
I know, but is a valid link in GitHub and other VCS sites.

Here's an example: <https://gist.github.com/64fe98f2d25c8faaeea11ebae7bf1014>

Perhaps a configuration to allow this kind of links?

**Comment by DavidAnson** [comment]
If you are saying any link fragment could potentially be valid, then you do not want any of them to trigger a warning and so you can disable rule MD051 for that file or for your project.

**Comment by anonymous1184** [comment]
I was thinking more about an exception for query fragments starting in `file-`, as most (if not all) of the VCS sites use it.

I basically only write in MD, so the extension is a **must** for me... super practical and highly configurable (I have projects with tabs, 2/4 spaces for indentation, some that require HTML tags, some that don't, etc...).

If there is no way to add it, I can ignore the warnings or disable it per-file (`<!-- markdownlint-disable-file MD051 -->`) once I'm sure all the other links are good to go.

Thanks a lot for this useful tool and your time.
